stylegan_code_finder/lightning_modules/trans_u_net_lightning.py

```diff
@@ -0,0 +1,62 @@
+from typing import List
+
+import numpy as np
+import torch
+from lightning_modules.base_lightning import BaseSegmenter
+from networks.trans_u_net.utils import DiceLoss
+from networks.trans_u_net.vit_seg_modeling import VisionTransformer, VIT_CONFIGS
+from torch import nn
+from torch.optim import Optimizer, SGD
+
+
+class TransUNetSegmenter(BaseSegmenter):
+    def __init__(self, configs: dict):
+        super().__init__(configs)
+        self.ce_loss = nn.CrossEntropyLoss()
+        self.dice_loss = DiceLoss(configs['num_classes'])
+        self.optimizers = self.get_optimizers()
+
+    def _initialize_segmentation_network(self):
+        transformer_config = VIT_CONFIGS[self.configs['pretrained_model_name']]
+        transformer_config.n_classes = self.configs['num_classes']
+        transformer_config.n_skip = self.configs['num_skip_channels']
+        vit_patches_size = self.configs['vit_patch_size']
+        transformer_config.patches.grid = (self.configs['image_size'] // vit_patches_size,
+                                           self.configs['image_size'] // vit_patches_size)
+
+        segmentation_network = VisionTransformer(transformer_config, img_size=self.configs['image_size'],
+                                                 num_classes=transformer_config.n_classes)
+        if self.configs['fine_tune'] is None:
+            segmentation_network.load_from(weights=np.load(self.configs['pretrained_path']))
+        self.segmentation_network = segmentation_network
+
+    def training_step(self, batch, batch_idx):
+        prediction = self.segmentation_network(batch['images'])
+
+        ground_truth = torch.squeeze(batch['segmented'], dim=1)
+        loss_ce = self.ce_loss(prediction, ground_truth.long())

```

Comment:
Loss calculation is pretty much the same for both steps. Would create a function.